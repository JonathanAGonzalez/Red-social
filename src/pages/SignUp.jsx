import { useState } from 'react';
import { SignUpValidation } from '../validation/validation';
import InputComponent from '../components/InputComponent';

const SignUp = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState({});


  const createAccount = async (e) => {
    e.preventDefault();
    setError({});
    try {
      const validation = SignUpValidation(email, password);
      if (validation.isValid) {
        setError(validation.errors);
      };


    } catch (error) {
      console.log(error);
    }

  }

  console.log(error);

  return (
    <div className="container pt-5">
      <div className="row">
        <div className="col-md-6 col-sm-10 mx-auto">
          <div className="card">
            <div className="card-body">
              <h3>Crear cuenta</h3>
              <hr />

              <form onSubmit={createAccount} id="form" >

                <InputComponent
                  labelText='Correo Electronico:'
                  id='email'
                  type='email'
                  placeholder='Jona@jona.com'
                  value={email}
                  onChange={setEmail}
                  error={error.email}
                />

                <InputComponent
                  labelText='Contraseña:'
                  id='password'
                  type='password'
                  placeholder='Ingresá tu contraseña'
                  value={password}
                  onChange={setPassword}
                  error={error.password}
                />

                <input className='btn btn-primary' aria-label='Crear cuenta' type="submit" value="Crear cuenta" id='submit' />
              </form>
            </div>
          </div>
        </div>
      </div>
    </div >
  );
};

export default SignUp;
