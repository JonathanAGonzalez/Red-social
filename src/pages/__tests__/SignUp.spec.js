import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SignUp from '../SignUp';

describe('SignUp', () => {
  describe('Layout', () => {
    test('Rendered correctly', () => {
      const { asFragment } = render(<SignUp />);
      expect(asFragment()).toMatchSnapshot();
    });

    test('It must have a field to write email.', () => {
      render(<SignUp />);
      const inputEmail = screen.getByLabelText('Correo Electronico:');
      expect(inputEmail).toBeInTheDocument();
    });

    test('It must have a field to write password.', () => {
      render(<SignUp />);
      const inputPassword = screen.getByLabelText('ContraseÃ±a:');
      expect(inputPassword).toBeInTheDocument();
    });

    test('It must have a field to Submit.', () => {
      render(<SignUp />);
      const inputSubmit = screen.getByLabelText('Crear cuenta');
      expect(inputSubmit).toBeInTheDocument();
    });
  });
});
