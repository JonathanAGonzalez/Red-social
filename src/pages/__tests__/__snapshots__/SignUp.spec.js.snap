// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Layout Rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="container pt-5"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 col-sm-10 mx-auto"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <h3>
              Crear cuenta
            </h3>
            <hr />
            <form
              id="form"
            >
              <div
                class="mb-3"
              >
                <label
                  aria-label="Correo Electronico:"
                  class="form-label"
                  for="email"
                  id="email"
                >
                  Correo Electronico:
                </label>
                <input
                  class="form-control"
                  id="email"
                  name="email"
                  placeholder="Jona@jona.com"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="mb-3"
              >
                <label
                  aria-label="Contrase침a:"
                  class="form-label"
                  for="password"
                  id="password"
                >
                  Contrase침a:
                </label>
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Ingres치 tu contrase침a"
                  type="password"
                  value=""
                />
              </div>
              <input
                aria-label="Crear cuenta"
                class="btn btn-primary"
                id="submit"
                type="submit"
                value="Crear cuenta"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
